src/main/java/bio/terra/dao/DataProjectDao.java

```diff
@@ -24,38 +27,107 @@ public DataProjectDao(DataRepoJdbcConfiguration jdbcConfiguration) {
         jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
     }
 
-    public UUID createDatasetDataProject(DatasetDataProjectSummary datasetDataProjectSummary) {
-        String sql = "INSERT INTO dataset_data_project (dataset_id, project_resource_id) VALUES " +
-            "(:dataset_id, :project_resource_id)";
+    // TODO: DRY up
+
+    public UUID createFileDataBucket(FileDataBucketSummary fileDataBucketSummary) {
```

Comment:
Can you explain why we need to record every file and its bucket in the metadata database?
That data implicitly exists: the file is in the bucket. It is also recorded in the filesystem data. Won't this table will be huge - having a row for every file in any dataset? And having another row to map the file to a project?

Having two copies of the same information builds in another place where we can get skew.

Reply:
Yep, all good points. Fixing now. I started this by making it consistent with how I did this for datasets and snapshots. I can clean those up as well -- but will start with files here.